daprdocs/content/en/developing-applications/building-blocks/pubsub/pubsub-overview.md

```diff
@@ -3,49 +3,64 @@ type: docs
+The specific implementation used by your service is pluggable and configured as a Dapr Pub/sub component at runtime. This approach removes the dependency from your service and, as a result, makes your service more portable and flexible to changes.
+
+When using the Pub/sub API in Dapr:
+
+1. Your service makes a network call to a Dapr Pub/sub building block, exposed as a sidecar.
+1. The Pub/sub building block makes calls into a Dapr Pub/sub component that encapsulates a specific message broker product.
+1. To receive topics, Dapr subscribes to the Dapr Pub/sub component on behalf of your service and delivers the messages to an endpoint when they arrive.
+
+In the diagram below, a "shipping" service and an "email" service have both subscribed to topics published by a "cart" service. Each service loads Pub/sub component configuration files that point to the same Pub/sub message bus component; for example: Redis Streams, NATS Streaming, Azure Service Bus, or GCP Pub/Sub.
 
 <img src="/images/pubsub-overview-components.png" width=1000>
 <br></br>
 
-The diagram below has the same services, however this time showing the Dapr publish API that sends an "order" topic and order endpoints on the subscribing services that these topic messages are delivered posted to by Dapr.
+In the diagram below, the Dapr API posts an "order" topic from the publishing "cart" service to "order" endpoints on the "shipping" and "email" subscribing services.
 
 <img src="/images/pubsub-overview-publish-API.png" width=1000>
 <br></br>
 
-## Features
-The pub/sub building block provides several features to your application.
+[View the complete list of Pub/sub components that Dapr supports]({{< ref supported-pubsub >}}).
+
+## Dapr Pub/sub API features
+
+The Pub/sub building block brings several features to your application.
 
 ### Cloud Events message format
 
-To enable message routing and to provide additional context with each message, Dapr uses the [CloudEvents 1.0 specification](https://github.com/cloudevents/spec/tree/v1.0) as its message format. Any message sent by an application to a topic using Dapr is automatically  "wrapped" in a Cloud Events envelope, using `Content-Type` header value for `datacontenttype` attribute.
+To enable message routing and provide additional context with each message, Dapr uses the [CloudEvents 1.0 specification](https://github.com/cloudevents/spec/tree/v1.0) as its message format. Any message sent by an application to a topic using Dapr is automatically wrapped in a Cloud Events envelope, using [`Content-Type` header value]({{< ref "pubsub-overview.md#content-types" >}}) for `datacontenttype` attribute.
 
 Dapr implements the following Cloud Events fields:
 
 * `id`
 * `source`
 * `specversion`
 * `type`
-* `datacontenttype` (Optional)
+* `datacontenttype` (optional)
 
 The following example shows an XML content in CloudEvent v1.0 serialized as JSON:
```

Comment:
I would include this as a better example saying.

"The following shows an example of a `orders` topic message sent by Dapr that includes a W3C `traceid` unique to the message, the `data` and the fields for the CloudEvent where the data content is serialized as JSON. 

{
  "topic": "orders",
  "pubsubname": "order_pub_sub",
  "traceid": "00-113ad9c4e42b27583ae98ba698d54255-e3743e35ff56f219-01",
  "tracestate": "",
  "data": {
    "orderId": 1
  },
  "id": "5929aaac-a5e2-4ca1-859c-edfe73f11565",
  "specversion": "1.0",
  "datacontenttype": "application/json; charset=utf-8",
  "source": "checkout",
  "type": "com.dapr.event.sent",
  "traceparent": "00-113ad9c4e42b27583ae98ba698d54255-e3743e35ff56f219-01"
}